crates/aptos/src/update/prover_dependencies.rs

```diff
@@ -0,0 +1,213 @@
+            } else {
+                return Err(CliError::UnexpectedError(format!(
+                    "{} is already set to a different value: {}.",
+                    env_var, current_value
+                )));
+            }
+        }
+
+        set_env::set(env_var, install_path.to_string_lossy())
+            .map_err(|e| CliError::UnexpectedError(format!("Failed to set {}: {}", env_var, e)))?;
+        println!(
+            "Added {} to environment with value: {}.",
+            env_var,
+            install_path.to_string_lossy()
+        );
+        if env::var(env_var).is_err() {
+            println!("Please use the `source` command or reboot the terminal to check whether {} is set with the correct value. \
+            If not, please set it manually.", env_var);
+        }
+        Ok(())
+    }
+
+    async fn download_dependency(&self) -> CliTypedResult<String> {
+        let build_info = cli_build_information();
+        let _ = match build_info.get(BUILD_OS).context("Failed to determine build info of current CLI")?.as_str() {
+            "linux-x86_64" => Ok("linux"),
+            "macos-aarch64" | "macos-x86_64" => Ok("macos"),
+            "windows-x86_64" => Ok("win"),
+            wildcard => Err(CliError::UnexpectedError(format!("Self-updating is not supported on your OS ({}) right now, please download the binary manually", wildcard))),
+        };
+
+        let install_dir = match self.install_dir.clone() {
+            Some(dir) => dir,
+            None => {
+                let dir = get_additional_binaries_dir();
+                // Make the directory if it doesn't already exist.
+                std::fs::create_dir_all(&dir)
+                    .with_context(|| format!("Failed to create directory: {:?}", dir))?;
+                dir
+            },
+        };
+
+        BoogieOptions::check_version_is_compatible(
+            BOOGIE_BINARY_NAME,
+            &format!("{}.0", TARGET_BOOGIE_VERSION),
+            MIN_BOOGIE_VERSION,
+            MAX_BOOGIE_VERSION,
+        )?;
+        let res = self
+            .install_binary(
+                install_dir.clone(),
+                BOOGIE_EXE,
+                BOOGIE_BINARY_NAME,
+                TARGET_BOOGIE_VERSION,
+                "/",
+                "Boogie program verifier version ",
+                BOOGIE_EXE_ENV,
+            )
+            .await?;
+        println!("{}", res);
+
+        BoogieOptions::check_version_is_compatible(
+            Z3_BINARY_NAME,
+            TARGET_Z3_VERSION,
+            MIN_Z3_VERSION,
+            MAX_Z3_VERSION,
+        )?;
+        let res = self
+            .install_binary(
+                install_dir.clone(),
+                Z3_EXE,
+                Z3_BINARY_NAME,
+                TARGET_Z3_VERSION,
+                "--",
+                "Z3 version ",
+                Z3_EXE_ENV,
+            )
+            .await?;
+        println!("{}", res);
+
+        #[cfg(unix)]
```

Comment:
So I guess Windows doesn't need this tool?

Reply:
Yeah, it is not supported windows in the current version, which may change in the future.